src/runtime/virtcontainers/pkg/cgroups/cgroups.go

```diff
@@ -116,13 +118,33 @@ func sandboxDevices() []specs.LinuxDeviceCgroup {
 
 func NewCgroup(path string, resources *specs.LinuxResources) (*Cgroup, error) {
 	var err error
+	var cgroup cgroups.Cgroup
 
 	cgroupPath, err := ValidCgroupPath(path, IsSystemdCgroup(path))
 	if err != nil {
 		return nil, err
 	}
 
-	cgroup, err := cgroups.New(cgroups.V1, cgroups.StaticPath(cgroupPath), resources)
+	if IsSystemdCgroup(path) {
+		slice, unit, err := getSliceAndUnit(path)
+		if err != nil {
+			return nil, err
+		}
+		// github.com/containerd/cgroups doesn't support creating a scope unit with
+		// v1 cgroups against systemd, the following intreacts directly with systemd
+		// to create the cgroup and then load it using containerd's api
+		err = createCgroupsSystemd(slice, unit, uint32(os.Getpid())) // adding runtime process, it makes calling setupCgroups redundant
+		// TODO: make pid as input variable?
```

Comment:
Is this comment redundant due to the `Getpid()` call above?

Reply:
Well, I meant through the function call, will remove though